app/misc/git.ts

```diff
@@ -1618,39 +1618,94 @@ function cleanRepo() {
 }
 
 /**
- * This method is called when the sync button is pressed, and causes the fetch-modal
- * to appear on the screen.
+ * This method is called when the user clicks the "Remote" button on the navbar.
  */
-function requestLinkModal() {
-  $("#fetch-modal").modal();
+function setUpstreamModal() {
+  $('#set-upstream-modal').modal('show');
+  let repository;
+  Git.Repository.open(repoFullPath)
+      .then(function (repo) {
+        repository = repo;
+        Git.Remote.lookup(repository, 'upstream').then(function(remote) {
+          document.getElementById("display-upstream").innerText = remote.url();
+        }, function(err) {
+          document.getElementById("display-upstream").innerText = 'No upstream repository currently configured.';
+        });
+    });
 }
 
 /**
- * This method is called when a valid URL is given via the fetch-modal, and runs the
- * series of git commands which fetch and merge from an upstream repository.
+ * Clears the fields from the upstream repo modal.
  */
-function fetchFromOrigin() {
-  console.log("begin fetching");
-  let upstreamRepoPath = document.getElementById("origin-path").value;
-  if (upstreamRepoPath != null) {
+function clearUpstreamModalText() {
+  document.getElementById("upstream-path").value = "";
+}
+
+/**
+ This function is called after the user enters the address of the upstream repo to sync from. It sets the
+ entered address as the upstream repo of the forked repository.
+ */
+function setUpstreamRepo() {
+  let repository;
+  let upstreamRepoPath = document.getElementById("upstream-path").value;
+  if(upstreamRepoPath != null) {
     Git.Repository.open(repoFullPath)
       .then(function (repo) {
-        console.log("fetch path valid")
-        displayModal("Beginning Synchronisation...");
-        addCommand("git remote add upstream " + upstreamRepoPath);
-        addCommand("git fetch upstream");
-        addCommand("git merge upstrean/master");
-        console.log("fetch successful")
-        updateModalText("Synchronisation Successful");
-        refreshAll(repo);
-      },
-        function (err) {
-          console.log("Waiting for repo to be initialised");
-          displayModal("Please select a valid repository");
-        });
-  } else {
-    displayModal("No Path Found.")
+      repository = repo;
+      var result = Git.Remote.createWithFetchspec(repository, 'upstream', upstreamRepoPath, '+refs/heads/*:refs/remotes/upstream/*');
+      addCommand("git remote add upstream " + upstreamRepoPath);
```

Comment:
You are adding the command twice (line 1656 and 1665). You should remove one?

Reply:
Note this doesn't occur with issue 133.

Reply:
I think we should remove the command from line 1656 and leave it at line 1665 after checking all the conditions.

Reply:
I think the addCommand line should be right before you issue the command to nodegit/simple-git. This way IF an error appears, then the user can see the last command issued and the applicable error. Make sense?

Reply:
I actually already solved this in PR #162. I deleted the duplicate command on line 1665 and replaced it with a "success" modal. This way the user knows that the command ran successfully.